docs/topics/hetzner-baremetal.md

```diff
@@ -0,0 +1,377 @@
+
+```bash
+$ clusterctl generate cluster my-cluster --list-variables --flavor hetzner-hcloud-control-planes
+Required Variables:
+  - HCLOUD_CONTROL_PLANE_MACHINE_TYPE
+  - HCLOUD_REGION
+  - HCLOUD_SSH_KEY
+  - HCLOUD_WORKER_MACHINE_TYPE
+
+Optional Variables:
+  - CLUSTER_NAME                 (defaults to my-cluster)
+  - CONTROL_PLANE_MACHINE_COUNT  (defaults to 3)
+  - KUBERNETES_VERSION           (defaults to v1.27.7)
+  - WORKER_MACHINE_COUNT         (defaults to 3)
+```
+
+These variables are used during the deployment of hetzner infrastructure provider in the cluster.
+
+Installing the hetzner provider can be using the following command: 
+```bash
+clusterctl init --infrastructure hetzner
+````
+
+```bash
+Fetching providers
+Installing cert-manager Version="v1.14.2"
+Waiting for cert-manager to be available...
+Installing Provider="cluster-api" Version="v1.6.3" TargetNamespace="capi-system"
+Installing Provider="bootstrap-kubeadm" Version="v1.6.3" TargetNamespace="capi-kubeadm-bootstrap-system"
+Installing Provider="control-plane-kubeadm" Version="v1.6.3" TargetNamespace="capi-kubeadm-control-plane-system"
+Installing Provider="infrastructure-hetzner" Version="v1.0.0-beta.33" TargetNamespace="caph-system"
+
+Your management cluster has been initialized successfully!
+
+You can now create your first workload cluster by running the following:
+
+  clusterctl generate cluster [name] --kubernetes-version [version] | kubectl apply -f -
+```
+
+### Generating Cluster Manifest
+
+Once the infrastructure provider is ready, we can create a cluster manifest using `clusterctl generate`
+
+```bash
+clusterctl generate cluster my-cluster --flavor hetzner-hcloud-control-planes > my-cluster.yaml
+```
+
+As of now, our cluster manifest lives in `my-cluster.yaml` file and we will apply this at a later stage after preparing secrets and ssh-keys.
+
+### Preparing Hetzner Robot
+
+1. Create a new web service user. [Here](https://robot.your-server.de/preferences/index), you can define a password and copy your user name.
+2. Generate an SSH key. You can either upload it via Hetzner Robot UI or just rely on the controller to upload a key that it does not find in the robot API. This is possible, as you have to store the public and private key together with the SSH key's name in a secret that the controller reads.
+
+For this tutorial, we will let the controller upload keys to hetzner robot.
+
+### Creating new user in Robot 
+To create new user in Robot, click on the `Create User` button in the Hetzner Robot console. Once you create the new user, a user ID will be provided to you via email from Hetzner Robot. The password will be the same that you used while creating the user.
+
+![robot user](../pics/robot_user.png)
+
+This is a required for following the next step.
+
+### creating and verify ssh-key in hcloud 
+```bash
+hcloud ssh-key create --name test --public-key-from-file ~/.ssh/test.pub
```

Comment:
what is "test"? I think you forgot a step where you generate an ssh key called "test"